src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.cs

```diff
@@ -55,34 +55,43 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             return builder.ToImmutable();
         }
 
-        private abstract class BaseJsonConverter : JsonConverter
+        private abstract class BaseJsonConverter<T> : JsonConverter
         {
-            protected static T ReadProperty<T>(JsonSerializer serializer, JsonReader reader)
+            public sealed override bool CanConvert(Type objectType) => typeof(T) == objectType;
+
+            public sealed override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+                => ReadValue(reader, objectType, existingValue, serializer);
+
+            public sealed override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+                => WriteValue(writer, (T)value, serializer);
+
+            protected abstract T ReadValue(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer);
```

Comment:
Does anyone use `objectType` or `existingValue`?  Consider removing to simplify if we don't need it?